physical/oci_objectstorage/oci_objectstorage.go

```diff
@@ -0,0 +1,398 @@
+	metricPutAcquirePool    = []string{"oci_objectstorage", "putAcquirePool"}
+
+	metricDeleteFailed         = []string{"oci_objectstorage", "deleteFailed"}
+	metricGetFailed            = []string{"oci_objectstorage", "getFailed"}
+	metricListFailed           = []string{"oci_objectstorage", "listFailed"}
+	metricPutFailed            = []string{"oci_objectstorage", "putFailed"}
+	metricHaWatchLockRetriable = []string{"oci_objectstorage", "haWatchLockRetriable"}
+	metricPermitsUsed          = []string{"oci_objectstorage", "permitsUsed"}
+
+	metric5xx = []string{"oci_objectstorage", "5xx"}
+)
+
+type Backend struct {
+	client         *objectstorage.ObjectStorageClient
+	bucketName     string
+	logger         log.Logger
+	permitPool     *physical.PermitPool
+	namespaceName  string
+	haEnabled      bool
+	lockBucketName string
+}
+
+func NewBackendConstructor(
+	client *objectstorage.ObjectStorageClient,
+	bucketName string,
+	logger log.Logger,
+	permitPool *physical.PermitPool,
+	namespaceName string,
+	haEnabled bool,
+	lockBucketName string) *Backend {
+	return &Backend{
+		client:         client,
+		bucketName:     bucketName,
+		logger:         logger,
+		permitPool:     permitPool,
+		namespaceName:  namespaceName,
+		haEnabled:      haEnabled,
+		lockBucketName: lockBucketName,
+	}
+}
+
+func NewBackend(conf map[string]string, logger log.Logger) (physical.Backend, error) {
+	bucketName := conf["bucketName"]
+	if bucketName == "" {
+		return nil, errors.New("missing bucket name")
+	}
+
+	namespaceName := conf["namespaceName"]
+	if bucketName == "" {
+		return nil, errors.New("missing namespace name")
+	}
+
+	lockBucketName := ""
+	haEnabled := false
+	var err error
+	haEnabledStr := conf["ha_enabled"]
+	if haEnabledStr != "" {
+		haEnabled, err = strconv.ParseBool(haEnabledStr)
+		if err != nil {
+			return nil, errwrap.Wrapf("failed to parse HA enabled: {{err}}", err)
+		}
+
+		if haEnabled {
+			lockBucketName = conf["lockBucketName"]
+			if lockBucketName == "" {
+				return nil, errors.New("missing lock bucket name")
+			}
+		}
+	}
+
+	authTypeAPIKeyBool := false
+	authTypeAPIKeyStr := conf["authTypeAPIKey"]
+	if authTypeAPIKeyStr != "" {
+		authTypeAPIKeyBool, err = strconv.ParseBool(authTypeAPIKeyStr)
+		if err != nil {
+			return nil, errwrap.Wrapf("failed parsing authTypeAPIKey parameter: {{err}}", err)
+		}
+	}
+
+	var cp common.ConfigurationProvider
+	if authTypeAPIKeyBool {
+		cp = common.DefaultConfigProvider()
+	} else {
+		cp, err = auth.InstancePrincipalConfigurationProvider()
+		if err != nil {
+			return nil, errwrap.Wrapf("failed creating InstancePrincipalConfigurationProvider: {{err}}", err)
+		}
+	}
+
+	objectStorageClient, err := objectstorage.NewObjectStorageClientWithConfigurationProvider(cp)
+	if err != nil {
+		return nil, errwrap.Wrapf("failed creating NewObjectStorageClientWithConfigurationProvider: {{err}}", err)
+	}
+
+	logger.Debug("physical/oci: configuration",
```

Comment:
You can omit the `physical/oci:` here and everywhere else. The central logger will prefix the message with this information.